src/create_shards.c

```diff
@@ -389,26 +394,38 @@ ParseWorkerNodeFile(char *workerNodeFilename)
 	}
 
 	/* build pattern to contain node name length limit */
-	snprintf(workerLinePattern, sizeof(workerLinePattern), "%%%us%%*[ \t]%%10u",
-			 MAX_NODE_LENGTH);
+	snprintf(workerLinePattern, sizeof(workerLinePattern), workerPatternTemplate,
+			 MAX_NODE_LENGTH, MAX_PORT_LENGTH);
 
 	while (fgets(workerNodeLine, sizeof(workerNodeLine), workerFileStream) != NULL)
 	{
+		const int workerLineLength = strnlen(workerNodeLine, MAXPGPATH);
 		WorkerNode *workerNode = NULL;
 		char *linePointer = NULL;
-		uint32 nodePort = 0;
-		int parsedValues = 0;
+		int32 nodePort = PostPortNumber; /* default port number */
+		int fieldCount = 0;
+		bool lineIsInvalid = false;
 		char nodeName[MAX_NODE_LENGTH + 1];
+		char nodePortString[MAX_PORT_LENGTH + 1];
 		memset(nodeName, '\0', sizeof(nodeName));
+		memset(nodePortString, '\0', sizeof(nodePortString));
 
-		if (strnlen(workerNodeLine, MAXPGPATH) == MAXPGPATH - 1)
+		if (workerLineLength == MAXPGPATH - 1)
 		{
 			ereport(ERROR, (errcode(ERRCODE_CONFIG_FILE_ERROR),
 							errmsg("worker node list file line exceeds the maximum "
 								   "length of %d", MAXPGPATH)));
 		}
 
-		/* skip leading whitespace and check for # comment */
+		/* trim trailing newlines preserved by fgets, if any */
+		linePointer = workerNodeLine + workerLineLength - 1;
+		while (linePointer >= workerNodeLine &&
```

Comment:
I initially _did_ do this, but changed it. When we print the line as part of the error message, it also printed the newline, which made a blank line in the final output, which I found annoying. By actually removing newlines from the end of `workerNodeLine` we can print it without having `\n\n` appear (PostgreSQL adds a line on its own, so ours is redundant).